aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -42,7 +54,9 @@ Tensor to(const Tensor& self, const TensorOptions& options, bool non_blocking, b
   }
   const auto & dtype_opt = options.dtype_opt();
   if ((!device_opt || self.device() == device_opt.value()) &&
-      (!dtype_opt  || self.dtype()  ==  dtype_opt.value()) && !copy) {
+      (!dtype_opt || self.dtype() == dtype_opt.value()) && !copy &&
+      optional_memory_format.value_or(MemoryFormat::Preserve) ==
+          MemoryFormat::Preserve) {
```

Comment:
Is this the only condition we should return self? Shouldn't we also return self if the requested memory format matches what the tensor is?